### PR TITLE
Adds long and double-click support to Button and Card

### PR DESCRIPTION
**Pull request by garanj**
(Credit to @yschimke for much of the work)

#### WHAT

Adds `onLongClick` and `onDoubleClick` support to `Card` and `Button`

#### WHY

These are not provided by default in the Wear Compose implementation, but are useful in a number of scenarios.

#### HOW

Custom `Card` and `Button` implementations, using `Box` to handle the long and double-click interactions.

#### Checklist :clipboard:
- [X] Add explicit visibility modifier and explicit return types for public declarations
- [X] Run spotless check
- [X] Run tests
- [X] Update metalava's signature text files

### CODE REVIEW

File: compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonA11yTest_default.png
Comment (yschimke) [inline compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonA11yTest_default.png:1]: Two "Role Button" are detected.

File: media/ui/src/test/snapshots/images/com.google.android.horologist.media.ui.screens.entity_PlaylistDownloadScreenA11yScreenshotTest_playlistDownloadScreenPreviewLoadedFullyDownloaded.png
Comment (yschimke) [inline media/ui/src/test/snapshots/images/com.google.android.horologist.media.ui.screens.entity_PlaylistDownloadScreenA11yScreenshotTest_playlistDownloadScreenPreviewLoadedFullyDownloaded.png:1]: A11y is split across 2 items.

File: compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonA11yTest_default.png
Comment (garanj) [reply compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonA11yTest_default.png:1]: Ah yes! Apologies

**Comment by garanj** [comment]
Have had a stab at improving the Compose semantics handling

File: compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonA11yTest_disabled.png
Comment (yschimke) [inline compose-material/src/test/snapshots/images/com.google.android.horologist.compose.material_ButtonA11yTest_disabled.png:1]: Missing disabled

File: compose-material/src/main/java/com/google/android/horologist/compose/material/Card.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        backgroundPainter = backgroundPainter,
+        contentColor = contentColor,
+        enabled = enabled,
+        contentPadding = contentPadding,
+        shape = shape,
+        interactionSource = interactionSource,
+        role = role,
+    ) {
+        Box(
+            modifier = Modifier
```
Comment (yschimke) [inline compose-material/src/main/java/com/google/android/horologist/compose/material/Card.kt:70]: Does the combined clickable go right to the edge with this?

Comment (yschimke) [reply compose-material/src/main/java/com/google/android/horologist/compose/material/Card.kt:70]: We should maybe confirm with a test, clicking inside the component.

Comment (garanj) [reply compose-material/src/main/java/com/google/android/horologist/compose/material/Card.kt:71]: > Does the combined clickable go right to the edge with this?

Nope! Nice. Tried to fix this now.

File: compose-material/src/main/java/com/google/android/horologist/compose/material/Card.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                .combinedClickable(
+                    interactionSource = interactionSource,
+                    indication = null,
+                    enabled = enabled,
+                    onClick = onClick,
+                    onLongClick = onLongClick,
+                    onDoubleClick = onDoubleClick,
+                    role = role,
+                )
+                .padding(contentPadding),
```
Comment (yschimke) [inline compose-material/src/main/java/com/google/android/horologist/compose/material/Card.kt:82]: Nice!

**Review by yschimke — APPROVED** [review]

**Pull request merged**